### PR TITLE
kernel-lmp-fitimage: add support for fpga

### PR DESCRIPTION
**Pull request by ricardosalveti**
Add support to store and load fpga binaries as part of the kernel
fitimage.

FPGA support depends on FPGA_BINARY and FPGA_LOADADDRESS, which needs to
be defined by the user.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>

### CODE REVIEW

**Comment by ricardosalveti** [comment]
To make it work reliably the layer will have to extend the linux recipe and add "do_compile[depends] += " fpga-recipe:do_deploy"", besides defining FPGA_BINARY and FPGA_LOADADDRESS.

File: classes/kernel-lmp-fitimage.bbclass [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
+FPGA_BINARY ?= "fpga.bin"
+
 #
 # Emit the fitImage ITS configuration section
 #
 # $1 ... .its filename
 # $2 ... Linux kernel ID
 # $3 ... DTB image name
 # $4 ... ramdisk ID
-# $5 ... config ID
```
Comment (mike-scott) [inline classes/kernel-lmp-fitimage.bbclass]: Are we dropping config ID? Or are we calling that Setup ID now?

Comment (ricardosalveti) [reply classes/kernel-lmp-fitimage.bbclass]: Original comment was wrong, $5 is actually setup id.

Comment (mike-scott) [reply classes/kernel-lmp-fitimage.bbclass]: ok

**Pull request merged**